controllers/gateway/operator.go

```diff
@@ -40,20 +47,25 @@ func newGatewayOperationCtx(gw *v1alpha1.Gateway, controller *GatewayController)
 }
 
 func (goc *gwOperationCtx) operate() error {
-	goc.log.Info().Msg("started operating on the gateway")
+	defer goc.persistUpdates()
+	goc.log.Info().Msg("operating on the gateway")
+
 	// validate the gateway
 	err := goc.validate()
 	if err != nil {
 		goc.log.Error().Err(err).Msg("gateway validation failed")
+		goc.markGatewayPhase(v1alpha1.NodePhaseError, "validation failed")
 		return err
 	}
-	gatewayClient := goc.controller.gatewayClientset.ArgoprojV1alpha1().Gateways(goc.gw.Namespace)
 
 	// manages states of a gateway
-	switch goc.gw.Status {
+	switch goc.gw.Status.Phase {
```

Comment:
thoughts on adding another phase `pending` since this may be relevant for situations after creation deployment + service but before creation of gateway deployment pod?

Reply:
makes sense.